grant-project/zcash-grant-system
# Prevent showing contribution instructions for canceled / expired proposals

Closes #323.

### What This Does

* Presents a pending contribution for a proposal that has failed or canceled as "Expired", preventing them from getting at the send instructions again. Not the most elegant fix, but quite simple.
* While in the neighborhood, fixes proposal cancels from trying to email anonymous contributions and erroring out.

### Steps to Test

* Create a new proposal
* Make an anonymous contribution to it, confirm it
* Make an auth'd contribution to it, leave it pending
* Cancel the proposal, confirm it worked without error
* Check the auth'd users contributions list, confirm that the contribution shows as "Expired"
